src/components/DefineCompany/BusinessContactInfo.vue

```diff
@@ -131,37 +180,52 @@ export default class BusinessContactInfo extends Mixins(CommonMixin) {
     return (this.contact.email === this.contact.confirmEmail) ? '' : 'Email addresses must match'
   }
 
-  // Life cycle methods
-  private mounted (): void {
-    if (this.showErrors) {
-      (this.$refs.form as Vue & { validate: () => boolean }).validate()
-    } else if (this.$refs.form && !this.isSame(this.initialValue, this.defaultBusinessContact)) {
-      (this.$refs.form as any).validate()
-    }
+  private get hasBusinessContactInfoChange () {
+    return !this.isSame(this.businessContact, this.getOriginalBusinessContact())
   }
 
   // Watchers
-  @Watch('initialValue', { deep: true, immediate: true })
+  @Watch('businessContact', { deep: true, immediate: true })
   private onContactPropValueChanged (): void {
-    this.contact = this.initialValue
+    this.contact = this.businessContact
+  }
+
+  private updateContactInfo (): void {
+    this.emitContactInfo(this.contact)
+    this.isEditing = false
+    this.emitHaveChanges(this.hasBusinessContactInfoChange)
+  }
+
+  private resetContactInfo (): void {
+    this.contact = this.getOriginalBusinessContact()
+    this.emitContactInfo(this.contact)
+    this.emitHaveChanges(false)
+    this.isEditing = false
+  }
+
+  private getOriginalBusinessContact (): BusinessContactIF {
```

Comment:
I think this can be a getter, which may improve performance a little.

Reply:
The reason why I made it like this is I did not want this component to have any dependency on the store. It should only have props and the wrapper in this case CorrectBusinessContactInfo and in case of alteration ( may be AlterBusinessContactInfo) should be the one interacting with store (read/write)

Reply:
I agree with your explanation but what I meant is that you could use the existing code and change it to a getter instead of a regular function.